include/circular_queue.h

```diff
@@ -6,10 +6,10 @@
 #include <utility>
 
 struct queue_elm {
-	bool dirty;    // is this queue element yet to be processed by sketching (if so do not overwrite)
-	bool touched;  // have we peeked at this item (if so do not peek it again)
-	uint32_t size; // the size of this data element (in bytes)
-	char *data;    // a pointer to the data
+	volatile bool dirty;    // is this queue element yet to be processed by sketching (if so do not overwrite)
```

Comment:
why not just have locking on the queue?

Reply:
We do. This was a bit of a throw things at the wall change. However, we have two separate mutexes protecting the queue so maybe there's something weird there.